### PR TITLE
BACK-958 : Lama service IT test

### PR DESCRIPTION
**Pull request by jjayet-ledger**
## What is this about?

This is testing the account registering, fetching, deleting and operations fetching.

## Cute picture of lama
44337b67caa020912aaf83c6dc5fb2f0

### CODE REVIEW

File: service/src/it/resources/application.conf [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "extended-public-key": "xpub6CMDks73kuNu3p827zgf6q6aNfpdeTLHFQsM4ssEBDMYHYJYHJ7tmMuTdZPk5456YZeoTaCRHNnbEK2GK5wkztQQCpVKEHb2tnm9N1gz1ZT"
+  "scheme": "BIP44",
+  "lookahead-size": 20,
+  "network": "MainNet",
+  "coin-family": "bitcoin",
+  "coin": "btc"
+}]
+
+server = {
+  host = "0.0.0.0"
```
Comment (nukky94) [inline service/src/it/resources/application.conf:11]: `localhost`

**Comment by estrauser-ledger** [comment]
LGTM

File: service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+class ServiceSpec extends AnyFlatSpecLike with Matchers {
+  implicit val cs: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+  implicit val t: Timer[IO]         = IO.timer(ExecutionContext.global)
+
+  val conf      = ConfigSource.default.loadOrThrow[ConfigSpec]
+  val serverUrl = s"http://${conf.server.host}:${conf.server.port}"
+
+  "Lama service" should "handle all kind of operations" in IOAssertion {
+    val accounts = Gen.someOf(conf.accounts).sample.get.toList
```
Comment (nukky94) [inline service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala:33]: just `conf.accounts`

File: service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    val accountRegisteringRequest =
+      Request[IO](method = Method.POST, uri = Uri.unsafeFromString(s"$serverUrl/account"))
```
Comment (nukky94) [inline service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala:35]: It was renamed to `/accounts` in plural

File: service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          BlazeClientBuilder[IO](global).resource
+            .use(_.expect[AccountRegistered](accountRegisteringRequest.withEntity(account)))
+
+        account <-
+          BlazeClientBuilder[IO](global).resource
+            .use(
+              _.expect[GetAccountManagerInfoResult](getAccountRequest(accountRegistered.accountId))
+            )
+
+        _ = Thread.sleep(60 * 1000)
```
Comment (nukky94) [inline service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala:64]: it's better to retry until there is some ops

File: service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    def removeAccountRequest(accountId: UUID) =
+      Request[IO](
+        method = Method.DELETE,
+        uri = Uri.unsafeFromString(s"$serverUrl/account/$accountId")
+      )
+
+    accounts.traverse { account =>
+      for {
+        accountRegistered <-
+          BlazeClientBuilder[IO](global).resource
```
Comment (nukky94) [inline service/src/it/scala/co/ledger/lama/service/ServiceSpec.scala:55]: Don't acquire and release the http client resource for each call.

You can do something like that:

```
BlazeClientBuilder[IO](global)
  .resource
  .use { client =>
     for {
        accountRegistered <- client.expect[AccountRegistered](accountRegisteringRequest.withEntity(account))
        ...
     } yield {...}
   }
```

**Review by nukky94 — CHANGES_REQUESTED** [review]

**Review by nukky94 — APPROVED** [review]

**Pull request merged**